functional_programming_tests/src/pathing_manager.cpp

```diff
@@ -18,33 +19,62 @@ Manager::Manager(std::unique_ptr<Manager::MiddlewareHandle> mw)
     : mw_{std::move(mw)} {
   mw_->register_set_map_service(
       [this](auto const request, auto response) -> void {
-        auto const path = utilities::parseSetMapRequest(request);
-        if (path) this->map_ = path.value();
-        response->success.data = path.has_value();
+
+        auto const set_map = [this](Map<unsigned char> const map)
+        -> tl::expected<Map<unsigned char>, utilities::parsing_set_map_error> {
+          this->map_ = map;
+          return map;
+        };
```

Comment:
I think you can just return `void`
```suggestion
        auto const set_map = [this](Map<unsigned char> const map)
        -> tl::expected<void, utilities::parsing_set_map_error> {
          this->map_ = map;
          return;
        };
```

Reply:
Nope, compiling fails